src/index.ts

```diff
@@ -1,28 +1,25 @@
-import * as ended from 'is-stream-ended';
 import {Transform} from 'stream';
 
 /**
  * Push an array of items into a Transform stream.
  * @param array The array you want to push to the stream.
  * @param stream The Transform stream into which array items are pushed.
  */
-export async function split(array: Array<{}>, stream: Transform) {
-  return new Promise<boolean>((resolve, reject) => {
-    const arr = [].slice.call(array);
-    function loopyloop() {
-      // Ensure the stream wasn't closed by the consumer.
-      const isEnded = ended(stream);
-      // Ensure al items from the array haven't been pushed.
-      const cont = !isEnded && arr.length > 0;
-      if (cont) {
-        stream.push(arr.shift());
-        // For large arrays, use setImmediate to ensure other microtasks
-        // and I/O operations have a chance to execute.
-        setImmediate(loopyloop);
-      } else {
-        resolve(isEnded);
+export async function split(
+    array: Array<{}>, stream: Transform): Promise<boolean> {
+  const arr = [].slice.call(array);
```

Comment:
I assume this line is to protect against passing of `arguments` in as a parameter.  Is this something we really need at this stage, now that `...params` are a thing?

Reply:
That is a good question, I just left this line as is. @stephenplusplus do you think we still need it?

Reply:
I did that to clone the user-provided array, so we're not modifying its contents.